### ISSUE TITLE
sendKeys and waitForElement

### ISSUE THREAD

**Issue by RamiroA92** [NativeScript/nativescript-dev-appium#43]
These properties waitForElement .. and sendKeys are no longer supported. 

For example, in the Nativescript Docs Plugin UI tests, it suggest to do the following: 

.waitForElementByXPath(passwordFieldElement, 20000).click()  
.sendKeys(PASSWORD)

 They are using version 1.0.0

How can we set the values of textfields in the new version ?

**Comment by SvetoslavTsenov** [comment]
Hi, 
Soon there is going to be a method for this, but for now, you can just refer to driver.driver. 
Basically when you create driver using the method createDriver you will receive our custom wrapper of appium and webdriverio. This class contains properties for appium driver and webdriver.

```
const driver: AppiumDriver = await createDriver();
driver.driver.waitForElementByXPath(passwordFieldElement, 20000).click()
.sendKeys(PASSWORD)

```

**Comment by RamiroA92** [comment]
Thank you, that was very helpful.

**Comment by gkvappium** [comment]
Hi @SvetoslavTsenov ,

In case I need to use complex gestures/actions (Which I do not see in the custom wrapper (AppiumDriver)), how do I achieve this?

Thanks

**Comment by SvetoslavTsenov** [comment]
Hey, @gkvappium I can point you to take a look at our implementation of scrolling where we use wd.TouchActions 
https://github.com/NativeScript/nativescript-dev-appium/blob/master/lib/appium-driver.ts#L234-L247. Basically, you can use driver.wd on your side to reach TouchActions object. I think that next week I will extend more actions and gestures support.
You can also read more on the touch action in [appium documentation](https://appium.readthedocs.io/en/stable/en/writing-running-appium/touch-actions/).
Let me know if need more info or suggestions for helper method that you currently need.

**Comment by gkvappium** [comment]
Thanks @SvetoslavTsenov . The thing is we already have a working python based framework for an existing app (We use toolium to provide out of the box page object modelling. This tool also provides various other features as well). I want to know few things before we start using nativescript-dev-appium for the new app that will be built on nativescript.

1. Are there any plugins/helper to provide out of the box page object model so as to isolate the UI from the functionality (Like toolium in python and page factory in java/selenium)? I couldn't find any for javascript unfortunately.
2. Does the tool generate any HTML reports? Or should something like mochawesome be integrated into this?
3. How to I override your appium wrapper and use the default appium javascript client in case any of the methods is missing or not working as expected in the wrapper?

Once again, thanks a lot for your quick response for my earlier post

**Comment by gkvappium** [comment]
@SvetoslavTsenov , can you please have a look into this ?

**Comment by SvetoslavTsenov** [comment]
Hi and apology for delayed answer but I wasn't available during that time. Let me start from your last question.
   3 . 'nativescript-dev-appium' plugin expose properties to webdriver, so basically you can use it to create your own implementations. Please have in mind that we accept pull request, so any missing implementation or bug could be revised and merged into the origin repo.
   2. The tool doesn't generate any type of report. Basically, we rely on the runner that will be used to run the tests to generate the report. For example, if you use mocha there are plenty of reports out of the box( XML, HTML and etc). We are planning to customize some HTML report to be able to show the results of image comparison in it. With that said you can use mocha-awesome just specify it within /config/mocha.opts file that the plugin creates on install.
 1. I don't know any javascript plugin which provides the functionality of toolium either but in my opinion, you can easily create a similar behaviour exposing by your own objects that you need. 

I hope that helps. Let me know if you need more information.

**Comment by gkvappium** [comment]
Thanks a lot @SvetoslavTsenov . This certainly helps. I've started the implementation and will surely get back to you in case I need any further assistance.
